File: app/src/main/java/au/com/digio/glidecropper/glide/CroppedImageModelLoader.java
package au.com.digio.glidecropper.glide;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.bumptech.glide.load.Options;
import com.bumptech.glide.load.model.ModelLoader;
import com.bumptech.glide.signature.ObjectKey;

public final class CroppedImageModelLoader implements ModelLoader<CroppedImage, CroppedImageDecoderInput> {

    private Context context;

    public CroppedImageModelLoader(Context context) {
        this.context = context;
    }

    @Nullable
    @Override
    public LoadData<CroppedImageDecoderInput> buildLoadData(@NonNull CroppedImage model,
                                                            int width,
                                                            int height,
                                                            @NonNull Options options) {
        return new LoadData<CroppedImageDecoderInput>(new ObjectKey(model),
        new CroppedImageDataFetcher(context, model));
    }

    @Override
    public boolean handles(@NonNull CroppedImage model) {
        return true;
    }
}

File: app/src/main/java/au/com/digio/glidecropper/glide/CroppedImageModelLoader.kt
package au.com.digio.glidecropper.glide

import android.content.res.Resources
import com.bumptech.glide.load.Options
import com.bumptech.glide.load.model.ModelLoader
import com.bumptech.glide.signature.ObjectKey

/**
 * Glide model loader to support the CroppedImage model.
 */
class CroppedImageModelLoader(val resources: Resources) : ModelLoader<CroppedImage, CroppedImageDecoderInput> {

    override fun handles(model: CroppedImage): Boolean = true

    override fun buildLoadData(
            model: CroppedImage,
            width: Int,
            height: Int,
            options: Options): ModelLoader.LoadData<CroppedImageDecoderInput>? {

        return ModelLoader.LoadData<CroppedImageDecoderInput>(
                ObjectKey(model),
                CroppedImageDataFetcher(resources, model))
    }
}